docs/registry/README.md

```diff
@@ -0,0 +1,385 @@
+```
+
+`3.` The `Event Consumer` creates a Trigger to listen to an EventType in the Registry. 
+
+Example:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: my-service-trigger
+  namespace: default
+spec:
+  filter:
+    sourceAndType:
+      type: dev.knative.foo.bar
+      source: dev.knative.example
+  subscriber:
+    ref:
+     apiVersion: serving.knative.dev/v1alpha1
+     kind: Service
+     name: my-service
+```
+
+### Broker Ingress Policies
+
+We briefly mentioned configuring an ingress policy in the Broker so that it can auto-register EventTypes. 
+In order to support that, we propose adding an `ingressPolicy` field to the Broker's spec CRD. 
+ 
+Below are two CR examples with Brokers configured using different policies. Note that the fields of 
+`ingressPolicy` might change (e.g., `ingressPolicy` might just end up being a string).
+We are omitting Broker's fields irrelevant to this discussion. 
+Also note that such configuration is done by the `Cluster Configurator`.
+
+- Allow Registered
+
+By setting the ingress policy to not allow any, the Broker will accept only events with EventTypes in the Registry.
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Broker
+metadata:
+  name: broker-allow-registered
+spec:
+  ingressPolicy:
+    allowAny: false
+```    
+
+- Auto Add
+
+By setting the ingress policy to auto add, the Broker will accept any event and will add its EventType 
+to the Registry (in case it is not present).
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Broker
+metadata:
+  name: broker-auto-add
+spec:
+  ingressPolicy:
+    autoAdd: true 
+```
+
+By not specifying an ingress policy, we thought that the default behavior would be to accept any event, which is similar to `Auto Add` 
+but without registration of events. This needs further discussion.
+
+Note that more policies should probably need to be configured in the future, e.g., allow auto-registration of EventTypes received 
+from within the cluster (e.g., from a response of a Service running in the cluster) as opposed to external services, and so on. 
+We just enumerate two simple cases here.
+
+## FAQ
+
+Here is a list of frequently asked questions that may help clarify the scope of the Registry.
+
+- Is the Registry meant to be used just for creating Triggers or for also setting up Event Sources?
+
+    It's mainly intended for helping the `Event Consumer` with the `discoverability` use case. Therefore, 
+    it is meant for helping out creating Triggers.
+
+- If I have a simple use case where I'm just setting up an Event Source and my KnService is its Sink (i.e., no Triggers involved), 
+is there a need/use for the Registry?
+
+    In this case, we believe there is no need for the Registry. As you can see in the EventType CRD, there is a mandatory 
+    `broker` field. If you are not sending events to a Broker, then there is no need to use a Registry. 
```

Comment:
Can you elaborate? I'm not clear on why a user not using a Broker/Trigger won't find value in a registry

Reply:
As defined for the MVP, the Registry defines the available Event Types that Triggers can be created for. Since there's no Broker, you don't create Triggers.

In this world you are manually instantiating a Source CR and instead of pointing it to a Broker, you're pointing it to somewhere else.
I think having an Event Source be a component that's usable outside of Eventing can be useful.

Reply:
I guess it might be useful. 
What I meant was that for this proposal, i.e., for the MVP, we only focus on Broker/Trigger. If you think that we need to add registry support for non-Broker sinks for the MVP, we can surely revisit this. Will rephrase the answer, as it can be confusing. Thanks for bringing that up.